lib/AST/ASTVerifier.cpp

```diff
@@ -1121,8 +1121,10 @@ class Verifier : public ASTWalker {
       verifyCheckedBase(S);
     }
     void verifyChecked(DeferStmt *S) {
-      auto FT = S->getTempDecl()->getInterfaceType()->castTo<AnyFunctionType>();
-      assert(FT->isNoEscape() && "Defer statements must not escape");
+      auto FT = S->getBody()->getType()->castTo<AnyFunctionType>();
+      // TODO: Making these into PBDs made them get marked as escaping... are
+      // there downstream effects?
+      // assert(FT->isNoEscape() && "Defer statements must not escape");
```

Comment:
Perhaps we could relax this assertion that if the defer body is not async then we keep the check that it is not escaping, but if it is it can be? Not sure if strictly necessary but we'd err on the side of caution this way 🤔